### ISSUE TITLE
Feature request: validation set callback

### ISSUE THREAD

**Issue by MattConflitti** [keras-team/keras#12356]
Hi!
So I have been researching [deepAR](https://docs.aws.amazon.com/sagemaker/latest/dg/deepar_how-it-works.html) from AWS and saw that they sample the time series in different places while training. I thought a parallel approach would be to be able to customize how Keras allows the validation set to be split. 

I was thinking that after each epoch, instead of just randomly shuffling (big no-no for time series data and LSTMs), allow a function to be called so that I can define where the split happens (randomly) and how big of chunk to take.

I think this would be more beneficial than just turning shuffle off and only validating on the last X% of the data.

Might seem like overkill, but I think there is a distinction for this. Any insight into how to do this currently would be appreciated!

```python
def customFunction(data):
    #get random number 0-1
    #split dataset at that location
    #extract chunk to validate on

model.fit(..., validation_callback=customFunction, ...)
```
